SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/EntityComponentTestUtils.h

```diff
@@ -0,0 +1,179 @@
+{
+
+namespace EntityComponentTestUtils
+{
+	static const Schema_FieldId EVENT_ID = 1;
+	static const Schema_FieldId EVENT_INT_FIELD_ID = 2;
+	static const Schema_FieldId TEST_DOUBLE_FIELD_ID = 1;
+} // namespace EntityComponentTestUtils
+
+inline ComponentData CreateTestComponentData(const Worker_ComponentId Id, const double Value)
+{
+	ComponentData Data{ Id };
+	Schema_Object* Fields = Data.GetFields();
+	Schema_AddDouble(Fields, EntityComponentTestUtils::TEST_DOUBLE_FIELD_ID, Value);
+	return Data;
+}
+
+inline ComponentUpdate CreateTestComponentUpdate(const Worker_ComponentId Id, const double Value)
+{
+	ComponentUpdate Update{ Id };
+	Schema_Object* Fields = Update.GetFields();
+	Schema_AddDouble(Fields, EntityComponentTestUtils::TEST_DOUBLE_FIELD_ID, Value);
+	return Update;
+}
+
+inline void AddTestEvent(ComponentUpdate* Update, int Value) {
+	Schema_Object* events = Update->GetEvents();
+	Schema_Object* eventData = Schema_AddObject(events, EntityComponentTestUtils::EVENT_ID);
+	Schema_AddInt32(eventData, EntityComponentTestUtils::EVENT_INT_FIELD_ID, Value);
+}
+
+inline ComponentUpdate CreateTestComponentEvent(const Worker_ComponentId Id, int Value) {
+	ComponentUpdate Update{ Id };
+	AddTestEvent(&Update, Value);
+	return Update;
+}
+
+/** Returns true if Lhs and Rhs have the same serialized form. */
+inline bool CompareSchemaObjects(const Schema_Object* Lhs, const Schema_Object* Rhs)
+{
+	const auto Length = Schema_GetWriteBufferLength(Lhs);
+	if (Schema_GetWriteBufferLength(Rhs) != Length)
+	{
+		return false;
+	}
+	const TUniquePtr<uint8_t[]> LhsBuffer = MakeUnique<uint8_t[]>(Length);
+	const TUniquePtr<uint8_t[]> RhsBuffer = MakeUnique<uint8_t[]>(Length);
+	Schema_SerializeToBuffer(Lhs, LhsBuffer.Get(), Length);
+	Schema_SerializeToBuffer(Rhs, RhsBuffer.Get(), Length);
+	return FMemory::Memcmp(LhsBuffer.Get(), RhsBuffer.Get(), Length) == 0;
+}
+
+/** Returns true if Lhs and Rhs have the same component ID and state. */
+inline bool CompareComponentData(const ComponentData& Lhs, const ComponentData& Rhs)
+{
+	if (Lhs.GetComponentId() != Rhs.GetComponentId())
+	{
+		return false;
+	}
+	return CompareSchemaObjects(Lhs.GetFields(), Rhs.GetFields());
+}
+
+/** Returns true if Lhs and Rhs have the same component ID and events. */
+inline bool CompareComponentUpdateEvents(const ComponentUpdate& Lhs, const ComponentUpdate& Rhs)
+{
+	if (Lhs.GetComponentId() != Rhs.GetComponentId())
+	{
+		return false;
+	}
+	return CompareSchemaObjects(Lhs.GetEvents(), Rhs.GetEvents());
+}
+
+/** Returns true if Lhs and Rhs have the same component ID and state. */
+inline bool CompareComponentUpdates(const ComponentUpdate& Lhs, const ComponentUpdate& Rhs)
+{
+	if (Lhs.GetComponentId() != Rhs.GetComponentId())
+	{
+		return false;
+	}
+	return CompareSchemaObjects(Lhs.GetFields(), Rhs.GetFields()) &&
+		CompareSchemaObjects(Lhs.GetEvents(), Rhs.GetEvents());
+}
+
+/** Returns true if Lhs and Rhs have the same entity ID, component ID, and state. */
```

Comment:
Nit: Not sure these comments are necessary

Reply:
Might be more useful it is says serialised state to make it clearer. I'd argue that the state part isn't entirely obvious when comparing schema_objects.
i.e. it's a compare by value for some transformation of the stored resource.
Rather than a compare by reference or internal state etc.